packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPluginCreateCameraTests.swift

```diff
@@ -0,0 +1,124 @@
+      globalAPI: MockGlobalEventApi(),
+      deviceDiscoverer: MockCameraDeviceDiscoverer(),
+      permissionManager: mockPermissionManager,
+      deviceFactory: { _ in MockCaptureDevice() },
+      captureSessionFactory: { mockCaptureSession },
+      captureDeviceInputFactory: MockCaptureDeviceInputFactory()
+    )
+
+    return (cameraPlugin, mockPermissionManager, mockCaptureSession)
+  }
+
+  func testCreateCamera_requestsOnlyCameraPermissionWithAudioDisabled() {
+    let (cameraPlugin, mockPermissionManager, _) = createSutAndMocks()
+    let expectation = expectation(description: "Initialization completed")
+
+    var requestCameraPermissionCalled = false
+    mockPermissionManager.requestCameraPermissionStub = { completion in
+      requestCameraPermissionCalled = true
+      completion?(nil)
+    }
+    var requestAudioPermissionCalled = false
+    mockPermissionManager.requestAudioPermissionStub = { completion in
+      requestAudioPermissionCalled = true
+      completion?(nil)
+    }
+
+    cameraPlugin.createCamera(
+      withName: "camera_name",
+      settings: FCPPlatformMediaSettings.make(
+        with: .medium,
+        framesPerSecond: nil,
+        videoBitrate: nil,
+        audioBitrate: nil,
+        enableAudio: false)
+    ) { result, error in
+      expectation.fulfill()
+    }
+
+    waitForExpectations(timeout: 30, handler: nil)
+
+    XCTAssertTrue(requestCameraPermissionCalled)
+    XCTAssertFalse(requestAudioPermissionCalled)
+  }
+
+  func testCreateCamera_requestsCameraAndAudioPermissionWithAudioEnabled() {
+    let (cameraPlugin, mockPermissionManager, _) = createSutAndMocks()
+    let expectation = expectation(description: "Initialization completed")
+
+    var requestCameraPermissionCalled = false
+    mockPermissionManager.requestCameraPermissionStub = { completion in
+      requestCameraPermissionCalled = true
+      completion?(nil)
+    }
+    var requestAudioPermissionCalled = false
+    mockPermissionManager.requestAudioPermissionStub = { completion in
+      requestAudioPermissionCalled = true
+      completion?(nil)
+    }
+
+    cameraPlugin.createCamera(
+      withName: "camera_name",
+      settings: FCPPlatformMediaSettings.make(
+        with: .medium,
+        framesPerSecond: nil,
+        videoBitrate: nil,
+        audioBitrate: nil,
+        enableAudio: true)
+    ) { result, error in
+      expectation.fulfill()
+    }
+
+    waitForExpectations(timeout: 30, handler: nil)
+
+    XCTAssertTrue(requestCameraPermissionCalled)
+    XCTAssertTrue(requestAudioPermissionCalled)
+  }
+
+  func testCreateCamera_assignsCamera() {
+    let (cameraPlugin, mockPermissionManager, mockCaptureSession) = createSutAndMocks()
+    let expectation = expectation(description: "Initialization completed")
+
+    mockPermissionManager.requestCameraPermissionStub = { completion in
+      completion?(nil)
+    }
+    mockPermissionManager.requestAudioPermissionStub = { completion in
+      completion?(nil)
```

Comment:
do you want to verify these completions are actually called?

Reply:
Not in this test - whether the permissions are requested properly is handled in the tests above. Here, we just need to call those completions so that the execution can continue as if the permissions were granted

Reply:
gotcha, can you add a comment saying passing nil means granting the permission? it isn't obvious when reading the code.